tests/_test_msui/test_mscolab.py

```diff
@@ -901,6 +904,39 @@ def test_profile_dialog(self, qtbot):
             critbox.assert_called_once()
         assert not self.window.mscolab.profile_dialog.gravatarLabel.pixmap().isNull()
 
+    def test_upload_and_fetch_profile_image(self, qtbot):
+        self._connect_to_mscolab(qtbot)
+        self._login(qtbot, self.userdata[0], self.userdata[2])
+        self.window.mscolab.profile_action.trigger()
+
+        initial_pixmap = self.window.mscolab.profile_dialog.gravatarLabel.pixmap().toImage()
+
+        # Creating a new image
+        new_image = Image.new('RGB', (64, 64), color='cyan')
+        img_byte_arr = io.BytesIO()
+        new_image.save(img_byte_arr, format='JPEG')
+        img_byte_arr.seek(0)
+
+        # Path where the image will be temporarily stored
+        temp_image_path = os.path.join(os.getcwd(), 'new_profile_image.jpg')
```

Comment:
Instead of writing to CWD this should use a temporary path. You can use the `tmp_path` pytest fixture to get one: https://docs.pytest.org/en/latest/reference/reference.html#tmp-path.

This would also make the explicit cleanup at the end unnecessary.